shared/middleware/files_api.rb

```diff
@@ -498,14 +504,16 @@ def files_put_file(encrypted_channel_id, filename, body)
       manifest_is_unchanged = false
     end
 
-    # if we're also deleting a file (on rename), remove it from the manifest
-    if params['delete']
+    # Handle any deletion request, unless a case-only rename is occuring.
+    if params['delete'] && params['delete'].downcase != downcased_filename
```

Comment:
Should this be comparing the escaped or unescaped versions of the filenames? Does it matter? Why?